service/history/queues/executable.go

```diff
@@ -231,6 +231,7 @@ func (e *executableImpl) Execute() (retErr error) {
 	priorityTaggedProvider.Counter(metrics.TaskRequests.GetMetricName()).Record(1)
 	priorityTaggedProvider.Timer(metrics.TaskScheduleLatency.GetMetricName()).Record(startTime.Sub(e.scheduledTime))
 
+	// todo: check why is this late error handling here
```

Comment:
yeah I should do some refactor on this. The idea here is to emit those metrics (request & latency) no matter what (there's also some other details related to clearing attempts). 

The return value from this `Execute()` method can be regard as execution metadata & execution errors where metadata is always available/valid even if the execution itself failed. 
I should change the result to a struct so that there's no separate error return value from the method call, which by convention means all other returns values should not be looked at.